File: backend/auto-trading-bot/src/main/java/com/trading/autotradingbot/entity/Trade.java
package com.trading.autotradingbot.entity;

import com.trading.autotradingbot.entity.enums.TradeAction;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Trade {
    private Long id;
    private Long accountId;
    private LocalDateTime timestamp;
    private String symbol;
    private TradeAction action; // Maps to 'BUY' or 'SELL'
    private BigDecimal quantity;
    private BigDecimal price;
    private BigDecimal fee;
    private BigDecimal profitLoss;
    private BigDecimal finalBalance;
    private String strategyName; // e.g. "RSI_Crossover"
}

File: frontend/src/types/dto/Trade.ts
export interface Trade {
    id: number;
    timestamp: string; // LocalDateTime
    symbol: string;
    action: 'BUY' | 'SELL';
    quantity: string;
    price: string;
    fee: string;
    profitLoss: string;
    finalBalance: string;
    strategyName: string;
}